tests/cupy_tests/indexing_tests/test_insert.py

```diff
@@ -155,3 +155,39 @@ def test_1darray(self, dtype):
             val = self._compute_val(xp)
             with pytest.raises(ValueError):
                 xp.fill_diagonal(a, val=val, wrap=self.wrap)
+
+
+@testing.parameterize(*testing.product({
+    'n': [2, 4, -3, 0],
+    'ndim': [2, 3, 1, 0, -2],
+}))
+@testing.gpu
+class TestDiagIndices(unittest.TestCase):
+
+    @testing.numpy_cupy_array_equal()
+    def test_diag_indices(self, xp):
+        return xp.diag_indices(self.n, self.ndim)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 3), (2, 2, 2)],
+}))
+@testing.gpu
+class TestDiagIndicesFrom(unittest.TestCase):
+
+    @testing.numpy_cupy_array_equal()
+    def test_diag_indices_from(self, xp):
+        arr = testing.shaped_arange(self.shape, xp)
+        return xp.diag_indices_from(arr)
+
+    def test_non_equal_dims(self):
+        for xp in (numpy, cupy):
+            arr = testing.shaped_arange((3, 5), xp)
+            with pytest.raises(ValueError):
+                xp.diag_indices_from(arr)
+
+    def test_1darray(self):
+        for xp in (numpy, cupy):
+            arr = testing.shaped_arange((5,), xp)
+            with pytest.raises(ValueError):
+                xp.diag_indices_from(arr)
```

Comment:
Should we write these tests in a separate class to remove unnecessary multiple runs of these tests while testing? I mean they don't use `self.shape` but still will run twice in this particular case?

Reply:
Yes